### PR TITLE
Removing community layouts from Zeal60

### PR DESCRIPTION
**Pull request by Wilba6582**
This was turned on without Travis checking that all the community layouts build. Some of the conflicts are non-trivial, like having action_function() defined in zeal60.c and in the layout. I will turn them on again after this is resolved.

### CODE REVIEW

File: keyboards/zeal60/rules.mk [hunk 1]
```diff
@@ -76,4 +76,6 @@ RAW_ENABLE = yes
 DYNAMIC_KEYMAP_ENABLE = yes
 CIE1931_CURVE = yes
 
-LAYOUTS = 60_ansi 60_iso 60_hhkb 60_ansi_split_bs_rshift
+# DO NOT turn on community layouts unless you actually make them all compile
+# and then @Wilba6582 in the PR, or I will find you and I will cut you.
```
Comment (noroadsleft) [inline keyboards/zeal60/rules.mk]: :rofl:

**Comment by drashna** [comment]
Honestly, I don't think it matters if a few layouts don't work.  The main thing is that the default layout works.  And as long as that works, other errors can be ignored.

The problem with expecting people to update their layouts for aboard that breaks them is threefold, actually:
1. A lot of these are "one shot" contributors.  
2. The folder name may not match the github username (and the history may not be easy to track)
3. That's a lot of work to dump on people, making sure their layout works for every possible compatible board.  Past, present and future.  

Additionally, I that the functionality that you're looking for for the action_function may be doable by using `TT`, keycode  the update_tri_layer_state function, and defining the tap toggle as 3.

**Comment by Wilba6582** [comment]
@drashna I'm not expecting people to update their layouts. That's the point of turning this feature off. It was off when I first committed, and it got turned on from someone else's PR (that would have had the same build errors I got when I first committed), and now Travis is going to fail every time I touch these files.

**Comment by noroadsleft** [comment]
For what it may be worth, I tracked down the history of the offending keymap. The user who submitted it only has two commits to qmk_firmware, in March of 2017. I looked at their GitHub account and their qmk_firmware fork hasn't been updated in over a year.

What's the feeling about removing keymaps that are unmaintained? I'm personally a believer in the Layouts feature, but if an unmaintained keymap is breaking that feature for a keyboard, perhaps the offending keymap should be removed from the repo.

**Comment by Talljoe** [comment]
I just went through a lot of work to update my keymap so that it works with the new Zeal60 in order to have the same layout across all of my boards. Removing the community support for Zeal60 will break two of my most-used boards and will make for a sad Christmas. My [pull request](https://github.com/qmk/qmk_firmware/pull/3910) passed CI, so I'm not sure why it's breaking now but I'm happy to look into it and make it work.

**Comment by noroadsleft** [comment]
@Talljoe I know Travis does different checks depending on what the PR updates. I'm not 100% on the particulars, but sometimes it will check everything, sometimes only a handful of keymaps, sometimes in between.

I do agree with @drashna that removing Community Layout support isn't the answer, at least not long-term. Perhaps we need to decide if community layout keymaps can/should be edited by others to improve functionality, so in this case, Wilba (or someone else) can update the offending keymap as needed.

**Comment by drashna** [comment]
@Talljoe unfortunately, travis passed because of *what* was changed.  And it did a global check, rather than keyboard specific checks.  So the default layout for the Zeal60 compiled fine, but it did not check any other layouts. 

You can see that here: https://travis-ci.org/qmk/qmk_firmware/builds/429798712#L1305

But this is EXACTLY why I mentioned to @Wilba6582 that Travis isn't always trustworthy.

**Comment by jackhumbert** [comment]
@Talljoe can you try adding the layout support to your personal `rules.mk`? If you can do that and support yours, I think we can merge this until the others are supported as well.

**Comment by Talljoe** [comment]
@jackhumbert I'm not sure what you mean by that.  I just tried created a keymap in `zeal60` with a `rules.mk` to add the layouts. It doesn't work because it conflicts with the `layouts` keymap of the same name. At any rate, I wasn't able to remove the layouts from `zeal60` and add it back in a personal keymap.

**Comment by Talljoe** [comment]
It really seems like the Zeal60 keyboard is trying to implement a couple of keycodes that aren't actually specific to the Zeal60. They should either be implemented in quantum core or in a keymap. Reusing the keymap extension points in the keyboard code breaks a well-known API convention and it is not the fault of unxmaal or any other keymap makers.

**Comment by Wilba6582** [comment]
@Talljoe Which keycodes are you referring to? If it's just the "triple tap" ones that require actions then yeah, I had to write that in the only way possible, which like much of QMK is extending/overriding/subverting the normal execution to do something "different". The fact that this is in the keyboard implementation, and conflicts with similar "overriding" in one user's keymap, is the problem, and TBH I don't think it's valid to say what a "user" can do, cannot be generalized up to the keyboard level so users of that keyboard can share that feature. Not everything has to be promoted up to quantum core and bloating the common code for everyone.

**Comment by skullydazed** [comment]
Can we close this one now? I think all the community layouts have been fixed?

**Pull request closed without merging**